Obsidian/WorldData/Generators/Overworld/ChunkBuilder.cs

```diff
@@ -109,16 +143,113 @@ internal static void Surface(GenHelper helper, Chunk chunk)
                         else
                         {
-                            chunk.SetBlock(x, y, z, BlocksRegistry.Stone);
+                            if (y <= 0)
+                            {
+                                chunk.SetBlock(x, y, z, BlocksRegistry.Deepslate);
+                            }
+                            else
+                            {
+                                chunk.SetBlock(x, y, z, BlocksRegistry.Stone);
+                            }
                         }
                     }
                 }
 
+                chunk.SetBlock(x, -64, z, BlocksRegistry.Bedrock);
                 chunk.Heightmaps[HeightmapType.WorldSurfaceWG].Set(x, z, terrainHeight);
             }
         }
     }
 
+    internal static bool GenerateOreCheck(int height, int OreType)
+    {
+        switch (OreType)
+        {
+            case 0:
+                if (height >= 0 && height <= 320)
+                {
+                    return true;
+                }
+                else
+                {
+                    return false;
+                }
+                break;
+            case 1:
+                if ((height >= -63 && height <= 72) || (height >= 80 && height <= 320))
+                {
+                    return true;
+                }
+                else
+                {
+                    return false;
+                }
+                break;
+            case 2:
+                if (height >= -16 && height <= 112)
+                {
+                    return true;
+                }
+                else
+                {
+                    return false;
+                }
+                break;
+            case 3:
+                if (height >= -63 && height <= 30)
+                {
+                    return true;
+                }
+                else
+                {
+                    return false;
+                }
+                break;
+            case 4:
+                if (height >= -63 && height <= 64)
+                {
+                    return true;
+                }
+                else
+                {
+                    return false;
+                }
+                break;
+            case 5:
+                if (height >= -63 && height <= 16)
+                {
+                    return true;
+                }
+                else
+                {
+                    return false;
+                }
+                break;
+            case 6:
+                if (height >= -16 && height <= 320)
+                {
+                    return true;
+                }
+                else
+                {
+                    return false;
+                }
+                break;
+            case 7:
+                if (height >= -63 && height <= 16)
+                {
+                    return true;
+                }
+                else
+                {
+                    return false;
+                }
```

Comment:
Just return the condition instead of using multiple if statements ^ ^

Would also use switch pattern matching expression e.x
```cs
public bool GenerateOreCheck(int height, OreType type) => type switch
{
     OreType.WhateverOre => condition,
     OreType.WhateverOre => condition,
    ...
    _ => true
};
```
You don't need to do this ^ specifically but it would make it a bit more readable.